### PR TITLE
Propagate exceptions from ServiceBase.OnStart.

### LINKED ISSUE #26280: ServiceBase doesn't propagate exceptions from OnStart
When a service fails to start, ServiceBase doesn't inform SCM and doesn't propagate the exception to the caller of ServiceBase.Run.

This bug has already been fixed in the full .NET Framework v4.7.1.

### PR DESCRIPTION
**Pull request by davidmatson**
Fixes #26280.

This fix follows the fix made in the .NET Framework itself. So:
Submission containing materials of a third party: Microsoft Corporation
(or maybe not, since I think technically I'm submitting on behalf of Microsoft Corporation in the first place, so perhaps there's no "third party".)

### CODE REVIEW

File: src/System.ServiceProcess.ServiceController/src/System/ServiceProcess/ServiceBase.cs [hunk 1]
```diff
@@ -875,8 +897,20 @@ public unsafe void ServiceMainCallback(int argCount, IntPtr argPointer)
...
                 ThreadPool.QueueUserWorkItem(new WaitCallback(this.ServiceQueuedMainCallback), args);
                 _startCompletedSignal.WaitOne();
+
+                if (_startFailedException != null)
+                {
+                    // Inform SCM that the service could not be started successfully.
+                    // (Unless the service has already provided another failure exit code)
+                    if (_status.win32ExitCode == 0)
+                    {
+                        status.win32ExitCode = ServiceControlStatus.ERROR_EXCEPTION_IN_SERVICE;
```
Comment (benaadams) [inline src/System.ServiceProcess.ServiceController/src/System/ServiceProcess/ServiceBase.cs]: `status` is undefined

Comment (davidmatson) [reply src/System.ServiceProcess.ServiceController/src/System/ServiceProcess/ServiceBase.cs]: Thanks; fixed

**Comment by danmosemsft** [comment]
Is a test feasible using TestService?

**Comment by davidmatson** [comment]
I suspect a test is feasible; I just haven't done that before. I don't believe a test exists on the .NET Framework side - I tested the change manually before checking in there. I squeezed some time to make this fix available in .NET Core but don't likely have the time to add a test myself currently.

**Comment by danmosemsft** [comment]
@Anipik can you please add a test to this branch before merging it. Should be easy to test that an exception is thrown. You can't really test the status code returned to SCM (service control manager) so that can be untested.

**Pull request closed without merging**